ref: add config --project flag and standardize the others (#2064)
iterative/dvc.org: content/docs/command-reference/config.md

## Synopsis

```usage

> usage: dvc config [-h] [--global | --system | --project | --local] [-q | -v] [-u]
>                   [-l] [name] [value]

Reviewer: Wrapping this at 72 chars...
```suggestion
usage: dvc config [-h] [--global | --system | --project | --local]
                  [-q | -v] [-u] [-l] [name] [value]
```

takes a config option `name` (a config section and a key, separated by a dot)
and its `value` (any valid alpha-numeric string generally).

When reading (e.g. `value` is not provided or `--list` is used), the values are
read from the system, global, repo and local configuration files by default, and
options `--system`, `--global`, `--repo`, and `--local` can be used to tell the
command to read from only that location.

When writing (e.g. `value` is specified or `--unset` is used), the new value is
written to the projects configuration file by default (`.dvc/config`), and
options `--system`, `--global` and `--local` can be used to tell the command to
write to that location (you can say `--repo` but that is the default). The
regular project's config file (`.dvc/config`) is meant to be tracked by Git and
should not contain sensitive and/or user-specific information (passwords, SSH
keys, etc). Use the `--local` command option (flag) instead, to set (or
override) secrets:

| Flag           | Priority | Config file location |

> | -------------- | -------- | -------------------- |

Reviewer: These tables probably need an update?

takes a config option `name` (a config section and a key, separated by a dot)
and its `value` (any valid alpha-numeric string generally).

When reading (e.g. `value` is not provided or `--list` is used), the values are

> read from the system, global, repo and local configuration files by default, and

Author: ```suggestion
read from the system, global, project and local configuration files by default, and
```

> When reading (e.g. `value` is not provided or `--list` is used), the values are
> read from the system, global, project and local configuration files by default,
> and options `--system`, `--global`, `--project`, and `--local` can be used to
> tell the command to read from only that location.

Reviewer: ```suggestion
When reading config options (no `value` is given or `--list` is used), the
values are read from the system, global, project and local configuration files
(in that order). The `--system`, `--global`, `--project`, and `--local` options
can be used to read from that configuration only.
```

> When writing (e.g. `value` is specified or `--unset` is used), the new value is
> written to the projects configuration file by default (`.dvc/config`), and
> options `--system`, `--global` and `--local` can be used to tell the command to
> write to that location (you can say `--project` but that is the default). The
> regular project's config file (`.dvc/config`) is meant to be tracked by Git and
> should not contain sensitive and/or user-specific information (passwords, SSH
> keys, etc). Use the `--local` command option (flag) instead, to set (or
> override) secrets:

Reviewer: ```suggestion
When writing (a `value` is given or `--unset` is used), the new value is written
to the project-level config file by default (`.dvc/config`). Options `--system`,
`--global` and `--local` can be used to write to that location instead.

⚠️ Note that `.dvc/config` is meant to be tracked by Git and should not contain
sensitive user info or secrets (passwords, SSH keys, etc). Use `--local` when in
doubt.
```

| Flag                          | Priority | Config file location |
| ----------------------------- | -------- | -------------------- |
| `--local`                     | 1        | `.dvc/config.local`  |
| None or `--project` (default) | 2        | `.dvc/config`        |

> The `--global` and `--system` flags are also available to set config options for
> multiple projects and users, respectively:

Reviewer: ```suggestion
The `--global` and `--system` flags are also useful to set config options for
multiple projects or users, respectively.
```

takes a config option `name` (a config section and a key, separated by a dot)
and its `value` (any valid alpha-numeric string generally).

> When reading config options (no `value` is given or `--list` is used), the
> values are read from the system, global, project and local configuration files
> (in that order). The `--system`, `--global`, `--project`, and `--local` options
> can be used to read from that configuration only.

Reviewer: Forgot a comma before... Committing this one too:
```suggestion
When reading config options (no `value` is given or `--list` is used), the
values are read from the system, global, project, and local config files (in
that order). The `--system`, `--global`, `--project`, and `--local` options can
be used to read from that configuration only.
```

- `-u`, `--unset` - remove a specified config option from a config file.

> - `--local` - use a Git-ignored local config file. This is useful when you need
>   to specify private config option values that you don't want to track and share
>   with Git (credentials, private locations, etc).

Reviewer: ```suggestion
- `--local` - use the (Git-ignored) local config file. This is useful when you
  need to specify private config option values that you don't want to track and
  share with Git (credentials, private locations, etc).
```

> - `--project` - use a project's config file (.dvc/config).

Reviewer: ```suggestion
- `--project` - use the project's config file (.dvc/config) only, when reading
  config values (this is the default when writing).
```

> - `--global` - use a global config file (e.g. `~/.config/dvc/config`) instead of
>   the project's `.dvc/config`. Useful to apply config options to all your
>   projects.

Reviewer: ```suggestion
- `--global` - use the global config file (e.g. `~/.config/dvc/config`). Useful
  to set config options for all of your projects.
```

> - `--system` - use a system config file (e.g. `/etc/dvc/config`) instead of
>   `.dvc/config`. Useful to apply config options to all the projects (all users)
>   in the machine. May require superuser access e.g.
>   `sudo dvc config --system ...` (Linux).

Reviewer: ```suggestion
- `--system` - use the system config file (e.g. `/etc/dvc/config`). Useful to
  set config options for all of the projects in the machine (all users). May
  require superuser access e.g. `sudo dvc config --system ...` (Linux).
```
